code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm

```diff
@@ -214,8 +214,18 @@
 	metabolization_rate = 0.5 * REAGENTS_METABOLISM
 	taste_description = "charged metal"
 	self_consuming = TRUE
+	process_flags = ORGANIC | SYNTHETIC
 	var/shock_timer = 0
+	var/empremoval = TRUE
 
+/datum/reagent/teslium/on_mob_metabolize(mob/living/L)
+	. = ..()
+	var/datum/component/empprotection/empproof = L.GetExactComponent(/datum/component/empprotection)
+	if(empproof)//only grant and remove emp protection if they didn't have it when drinking it
+		empremoval = FALSE
+	else
+		L.AddComponent(/datum/component/empprotection, EMP_PROTECT_SELF)
```

Comment:
```suggestion
	if(!empproof) // only grant and remove EMP protection if they didn't have it when drinking it
		empremoval = TRUE
		L.AddComponent(/datum/component/empprotection, EMP_PROTECT_SELF)
```

If you do !empproof instead of empproof you get to remove the else statement